internal/notification/smtp_login_auth.go

```diff
@@ -2,20 +2,28 @@ package notification
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
 	"net/smtp"
 )
 
 type loginAuth struct {
 	username string
 	password string
+	host     string
 }
 
-func LoginAuth(username, password string) smtp.Auth {
-	return &loginAuth{username, password}
+func LoginAuth(username, password, host string) smtp.Auth {
+	return &loginAuth{username, password, host}
 }
 
 func (a *loginAuth) Start(server *smtp.ServerInfo) (string, []byte, error) {
```

Comment:
Can you convert it into a private function? `start(...)` instead of `Start(...)`.

Reply:
Can you also please add unit tests for this function?

Reply:
Making this private is not possible due to the interface which we have to implement: https://golang.org/src/net/smtp/auth.go?s=292:1191#L15

I'll have to figure out how to do the unit tests. Are the files under ./internal/suites/ a good place to start?

Reply:
The unit tests are in `internal/*/*_test.go` like the following examples:
- https://github.com/authelia/authelia/blob/master/internal/authorization/authorizer_test.go
- https://github.com/authelia/authelia/blob/master/internal/authentication/file_user_provider_test.go
- https://github.com/authelia/authelia/blob/master/internal/authentication/ldap_user_provider_test.go
- https://github.com/authelia/authelia/blob/master/internal/authentication/password_hash_test.go

```
INFO[0136] =====> Unit testing stage <=====
?   	github.com/authelia/authelia/cmd/authelia	[no test files]
?   	github.com/authelia/authelia/cmd/authelia-scripts	[no test files]
ok  	github.com/authelia/authelia/internal/authentication	(cached)
ok  	github.com/authelia/authelia/internal/authorization	(cached)
ok  	github.com/authelia/authelia/internal/configuration	(cached)
ok  	github.com/authelia/authelia/internal/configuration/schema	(cached)
ok  	github.com/authelia/authelia/internal/configuration/validator	(cached)
?   	github.com/authelia/authelia/internal/duo	[no test files]
ok  	github.com/authelia/authelia/internal/handlers	(cached)
?   	github.com/authelia/authelia/internal/logging	[no test files]
ok  	github.com/authelia/authelia/internal/middlewares	(cached)
?   	github.com/authelia/authelia/internal/mocks	[no test files]
?   	github.com/authelia/authelia/internal/models	[no test files]
?   	github.com/authelia/authelia/internal/notification	[no test files]
ok  	github.com/authelia/authelia/internal/regulation	(cached)
?   	github.com/authelia/authelia/internal/server	[no test files]
ok  	github.com/authelia/authelia/internal/session	(cached)
?   	github.com/authelia/authelia/internal/session/mocks	[no test files]
?   	github.com/authelia/authelia/internal/storage	[no test files]
?   	github.com/authelia/authelia/internal/templates	[no test files]
?   	github.com/authelia/authelia/internal/utils	[no test files]
```

You should be able to see each of the ones above to give you an idea.

Reply:
Thanks, that's really helpful! I'll see what I can do.

Reply:
Added to my branch. It tests most situations.